Core/Utilities/TimeLeft/TimeLeft.py

```diff
@@ -76,10 +85,23 @@ def getTimeLeft( self, cpuConsumed = 0.0 ):
       return resourceDict
 
     resources = resourceDict['Value']
-    if not resources['CPULimit'] or not resources['WallClockLimit']:
+    self.log.verbose( resources )
+    if not resources['CPULimit'] and not resources['WallClockLimit']:
       # This should never happen
       return S_ERROR( 'No CPU or WallClock limit obtained' )
 
+    # if one of CPULimit or WallClockLimit is missing, compute a reasonable value
+    if not resources['CPULimit']:
+      resources['CPULimit'] = resources['WallClockLimit']
```

Comment:
I agree, but we had recently an example where this was causing problems as the site had set CPULimit = 0.7*WallClockLimit but for a strange reason the CPULimit was not exposed :(
Jobs were killed by CPU limit then as it was reached before the WallClockLimit was reached... I agree it would be a hack rather than a real fix for a missing value as the factor would be random... An alternative is to just return an error if CPULimit is not provided by WallClockLimit is.